packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -249,33 +335,45 @@ struct SimState {
     msg_schema: Arc<Schema>,
 }
 
-struct RunnerImpl<'m> {
-    embedded: Embedded<'m>,
-    this: mv8::Value<'m>,
+struct ThreadLocalRunner<'s> {
+    embedded: Embedded<'s>,
+    this: v8::Local<'s, v8::Value>,
     sims_state: HashMap<SimulationShortId, SimState>,
 }
 
-// we pass in _mv8 for the return values lifetime
-fn sim_id_to_js(_mv8: &MiniV8, sim_id: SimulationShortId) -> mv8::Value<'_> {
-    mv8::Value::Number(sim_id as f64)
+fn sim_id_to_js<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    sim_id: SimulationShortId,
+) -> v8::Local<'s, v8::Value> {
+    v8::Number::new(scope, sim_id as f64).into()
 }
 
-// we pass in _mv8 for the return values lifetime
-fn pkg_id_to_js(_mv8: &MiniV8, pkg_id: PackageId) -> mv8::Value<'_> {
-    mv8::Value::Number(pkg_id.as_usize() as f64)
+fn pkg_id_to_js<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    pkg_id: PackageId,
+) -> v8::Local<'s, v8::Value> {
+    v8::Number::new(scope, pkg_id.as_usize() as f64).into()
 }
 
-fn idxs_to_js<'m>(mv8: &'m MiniV8, idxs: &[usize]) -> Result<mv8::Value<'m>> {
-    let a = mv8.create_array();
-    for (i, idx) in idxs.iter().enumerate() {
-        a.set(i as u32, mv8::Value::Number(*idx as u32 as f64))?;
+fn new_js_array_from_usizes<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    values: &[usize],
+) -> Result<v8::Local<'s, v8::Value>> {
+    let a = v8::Array::new(scope, values.len() as i32);
+    for (i, idx) in values.iter().enumerate() {
+        let js_idx = v8::Number::new(scope, *idx as u32 as f64);
+        a.set_index(scope, i as u32, js_idx.into())
+            .ok_or_else(|| Error::V8(format!("Could not set index {idx} on idxs_to_js Array")))?;
```

Comment:
This is pretty unclear as well. It's "Couldn't set value at index {idx} on JS array"

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/2a832f8dc26357b06320aed8e66cb59822d849a2